engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-associations.controller.js

```diff
@@ -31,12 +31,18 @@ angular.module('Bastion.activation-keys').controller('ActivationKeyAssociationsC
             $scope.table.working = false;
         }
 
-        contentHostsNutupane = new Nutupane(ActivationKey, params, 'contentHosts');
+        contentHostsNutupane = new Nutupane(Host, params);
+        contentHostsNutupane.searchTransform = function (term) {
+            var addition = "activation_key_id=" + $scope.$stateParams.activationKeyId;
+            if (term === "" || angular.isUndefined(term)) {
+                return addition;
+            }
+        };
```

Comment:
Looks like if I visit this page and then search for something else like 'name = foo'  its returning all results.  

I think you're missing the 
```
return term + " and " + addition;
```
at the end